nodeup/pkg/model/tests/golden/warm-pool/cluster.yaml

```diff
@@ -0,0 +1,83 @@
+apiVersion: kops.k8s.io/v1alpha2
+kind: Cluster
+metadata:
+  name: minimal.example.com
+spec:
+  kubernetesApiAccess:
+  - 0.0.0.0/0
+  channel: stable
+  cloudProvider: aws
+  configBase: memfs://clusters.example.com/minimal.example.com
+  etcdClusters:
+  - cpuRequest: 200m
+    etcdMembers:
+    - instanceGroup: master-us-test-1a
+      name: us-test-1a
+    memoryRequest: 100Mi
+    name: main
+    provider: Manager
+    backups:
+      backupStore: memfs://clusters.example.com/minimal.example.com/backups/etcd-main
+  - cpuRequest: 100m
+    etcdMembers:
+    - instanceGroup: master-us-test-1a
+      name: us-test-1a
+    memoryRequest: 100Mi
+    name: events
+    provider: Manager
+    backups:
+      backupStore: memfs://clusters.example.com/minimal.example.com/backups/etcd-events
+  iam: {}
+  kubelet:
+    anonymousAuth: false
+  kubernetesVersion: v1.18.0
+  masterInternalName: api.internal.minimal.example.com
+  masterPublicName: api.minimal.example.com
+  networkCIDR: 172.20.0.0/16
+  networking:
+    kubenet: {}
+  nonMasqueradeCIDR: 100.64.0.0/10
+  sshAccess:
+  - 0.0.0.0/0
+  topology:
+    masters: public
+    nodes: public
+  subnets:
+  - cidr: 172.20.32.0/19
+    name: us-test-1a
+    type: Public
+    zone: us-test-1a
+  warmPool:
+    images:
+    - docker.io/mysql:8
+    - docker.io/phpmyadmin:5
+
+---
+
+apiVersion: kops.k8s.io/v1alpha2
+kind: InstanceGroup
+metadata:
+  name: warmpool-us-test-1a
+  labels:
+    kops.k8s.io/cluster: minimal.example.com
+spec:
+  associatePublicIp: true
+  image: ami-1234
+  machineType: m3.medium
+  maxSize: 1
+  minSize: 1
+  role: Node
+  subnets:
+  - us-test-1a
+  warmPool:
+    images:
+    - coredns/coredns:1.8.3
+    - k8s.gcr.io/cpa/cluster-proportional-autoscaler:1.8.3
+    - k8s.gcr.io/etcdadm/etcd-manager:3.0.20210430
+    - k8s.gcr.io/kops/dns-controller:1.22.0-alpha.1
+    - k8s.gcr.io/kops/kops-controller:1.22.0-alpha.1
+    - k8s.gcr.io/kops/kube-apiserver-healthcheck:1.22.0-alpha.1
+    - k8s.gcr.io/kube-apiserver:v1.21.0
+    - k8s.gcr.io/kube-controller-manager:v1.21.0
+    - k8s.gcr.io/kube-proxy:v1.21.0
+    - k8s.gcr.io/kube-scheduler:v1.21.0
```

Comment:
You should be able to remove all these and the list of tasks in the next file would be preserved, correct? That would confirm that users don't need to specify them in their IG specs and nodeup will still prepull the relevant images from assets.

Reply:
Not really. The test harness is not doing a full mock from the spec, like the integration test does.
Somewhere there's a missing piece, but did not have time to dig that deep.

Reply:
ah ok. so if we had the integration tests also generate nodeup task lists, we could see the functionality there.

Reply:
would be cool to generate the nodeup tasks in same place, or in a similar way here.